File: programmers/js/basic/181858.js
// 무작위로 K개의 수 뽑기
function solution(arr, k) {
  const result = [];

  for (const num of new Set(arr)) {
    if (result.length >= k) {
      break;
    }

    result.push(num);
  }

  return result.length < k
    ? [...result, ...Array(k - result.length).fill(-1)]
    : result;
}

// 다른 풀이
// function solution(arr, k) {
//   const set = new Set(arr);
//   return set.size < k
//     ? [...set, ...Array(k - set.size).fill(-1)]
//     : [...set].slice(0, k);
// }

File: programmers/java/basic/181858.java
// 무작위로 K개의 수 뽑기

import java.util.ArrayList;

class Solution {
    public int[] solution(int[] arr, int k) {
        ArrayList<Integer> result = new ArrayList<>();
        
        for (int num : arr) {
            if (!result.contains(num)) {
                result.add(num);
            }
            
            if (result.size() == k) break;
        }
        
        while (result.size() < k) {
            result.add(-1);
        }
        
        return result.stream().mapToInt(v -> v).toArray();
    }
}